bindings/java/java/src/main/java/org/ethereum/evmc/Host.java

```diff
@@ -70,16 +70,17 @@ static ByteBuffer get_code_hash(int context_index, byte[] address) {
   }
 
   /** Copy code callback function. */
-  static ByteBuffer copy_code(int context_index, byte[] address, int code_offset) {
+  static ByteBuffer copy_code(int context_index, byte[] address) {
     HostContext context =
         requireNonNull(
             getContext(context_index),
             "HostContext does not exist for context_index: " + context_index);
     byte[] code = context.getCode(address).array();
 
-    if (code != null && code_offset > 0 && code_offset < code.length) {
-      int length = code.length - code_offset;
-      return ByteBuffer.allocateDirect(length).put(code, code_offset, length);
+    /* Ensure code is returned and both offset/length are positive numbers. */
+    if (code != null) {
+      // TODO: use wrap here, no need to duplicate this given it is copied again.
```

Comment:
Not sure I understand the TODO.

Reply:
To be less terse: this TODO says we can use `ByteBuffer.wrap`, am I reading this right? But for C code to interact with ByteBuffer, you need to use a direct memory allocation. I had panics when I tried to use heap-based bytebuffers.